### ISSUE TITLE
[RFE] Provide a way to reduce the amount of generated output for `doc` inspection.

### ISSUE THREAD

**Issue by jchecahi** [rpminspect/rpminspect#1233]
**Is your feature request related to a problem? Please describe.**
We are having issues with doc inspection in rust, which generates hundreds of thousands of lines due to minor changes in HTML documentation.

* Bodhi link: https://bodhi.fedoraproject.org/updates/FEDORA-2023-6f2c7aa713
* rpminspect run: https://artifacts.dev.testing-farm.io/f4848c12-9780-49d3-a251-a2b96fd86190/

This results in a results.json file of 1.8GB size, which makes almost impossible to review the results and in the worst case, can crash the pipeline.

**Describe the solution you'd like**
It would be great to have a way to reduce the amount of output generated by the `doc` inspection, so when rpminspect produces too much results, the output is still manageable. Some ideas to implement this could be:

* Being able to set a suppression threshold (similar to `rpminspect-fedora -s VERIFY <nvr>`) per inspection, so when we know a given inspection generates too many INFO results we can filter that without affecting the rest of the inspections. That would be extensible to other inspections.
* Truncate the diff output when it's too big, or behave like `cmp` does, which only indicates where two files differ.
* Print only the word changes in a given line, similar to what `git diff --word-diff=porcelain` does.
* Any other idea you could think of :)

**Describe alternatives you've considered**
We could exclude those HTML files from the inspection, but that's inconvenient as it can mask real issues.

**Additional context**
We have had issues in the past with inspections which dumps too much output, and can lead to other issues like segfaults or OOM kills. E.g.: https://github.com/rpminspect/rpminspect/pull/1181

**Comment by dcantrell** [comment]
I like these suggestions.  Let me think about this some more and see what ideas I can come up with.

**Comment by dcantrell** [comment]
OK, looking in to this one I came up with what I think is a simple solution.

First, the doc inspection in rpminspect always reports content changes at the INFO level.  The only time it is reported as VERIFY is if the builds being compared are not a rebase build and the spec file either gained or lost something marked as %doc.

So what I did was change the general reporting of content changes to exclude the diff output if the builds being compared are a rebase.  If the builds being compared are not a rebase build (that is, something like rust-1.72.0-1 compared to rust-1.72.0-2) then it will include the diff output.  To me, the latter case might be one where you want more visual indication of the changes (not necessarily for Fedora, but for RHEL maybe).

For testing I compared rust-1.71.1-1.fc38 to rust-1.73.0-1.fc38.  Before changes and just running the doc inspection, the results output in text mode use 3.4G.  After I made the changes above, the results output use 47M.  It is still generating a line of output for each file that's different, but I think that is ok.
